Framework/DataHandling/test/ISISDataArchiveTest.h

```diff
@@ -16,22 +16,71 @@
 using namespace Mantid::DataHandling;
 using namespace Mantid::API;
 
+namespace {
+#ifdef _WIN32
+const char *URL_PREFIX = "http://data.isis.rl.ac.uk/where.py/windir?name=";
+#else
+const char *URL_PREFIX = "http://data.isis.rl.ac.uk/where.py/unixdir?name=";
+#endif
+}
+
+class MockgetCorrectExtension : public ISISDataArchive {
+public:
+  std::string
+  getCorrectExtension(const std::string &path,
+                      const std::vector<std::string> &exts) const override {
+    (void)exts;
+    return path;
+  }
+};
+
 class ISISDataArchiveTest : public CxxTest::TestSuite {
 public:
-  void xtestSearch() {
-    ISISDataArchive arch;
+  /**
```

Comment:
Could we think about changing the `ISISDataArchive` class slightly so that we put in a protected virtual method, something like `sendRequest()` that is the one that deals with sending out the request to the webserver and then we just mock that out in here and leave everything else the same? The mock can then be controlled to return local paths that we could create in temp directories to check the logic?